### PR TITLE
[WIP] MAID-2526 refactor/webfetch: move webFetch to bg process, create reducer, create action

### PR DESCRIPTION
**Pull request by hunterlester**

### CODE REVIEW

File: app/background.process.development.js [hunk 1]
```diff
@@ -59,7 +61,8 @@ const init = async ( ) =>
     // Add middleware from extensions here. TODO: this should be be unified somewhere.
     // const loadMiddlewarePackages = [];
     const store = configureStore( {}, [], true );
-
+    // NOTE: redux store is an empty object here
+    logger.info('bg redux store:', store);
```
Comment (joshuef) [inline app/background.process.development.js]: if you want to see the state of the store, you need to `store.getState()`

File: app/background.process.development.js [hunk 2]
```diff
@@ -19,9 +19,11 @@ import manageRemoteCalls from './background.manageRemoteCalls';
 import sysUri from 'extensions/safe/ffi/sys_uri';
 import { loadExtensions }  from './extensions';
 import { setupServerVars, startServer } from './server';
+import { remote } from 'electron';
 
 import * as ffiLoader from 'extensions/safe/auth-api/ffiLoader';
 
+const store = remote.getCurrentWindow().reduxStore;
```
Comment (joshuef) [inline app/background.process.development.js]: store is setup on line 63

File: app/setupBackground.js [hunk 3]
```diff
@@ -5,52 +5,61 @@ import { isRunningUnpacked, isRunningDevelopment } from 'appConstants';
...
         } );
 
         backgroundProcess.loadURL( `file://${BACKGROUND_PROCESS}` );
     }
-
-    return backgroundProcess;
-};
+    else
+    {
+        resolve( backgroundProcess );
```
Comment (joshuef) [inline app/setupBackground.js]: i think you want this inside of the `did-finish-load` callback above, @hunterlester

File: app/background.process.development.js [hunk 4]
```diff
@@ -29,7 +32,7 @@ const manageLibLoading = ( store ) =>
     const listenerState = store.getState();
     const islistenerMock = listenerState.peruseApp.isMock;
 
-    if( islistenerMock === mockStatus )
+    if ( islistenerMock === mockStatus )
```
Comment (joshuef) [inline app/background.process.development.js]: @hunterlester do you think we should move `isMock` to the `authenticator` reducer? Does it make more sense there?

File: app/background.process.development.js [hunk 5]
```diff
@@ -17,6 +17,9 @@ import { initAnon, initMock } from 'extensions/safe/network';
 import { setIPCStore } from 'extensions/safe/ffi/ipc';
 import manageRemoteCalls from './background.manageRemoteCalls';
 import sysUri from 'extensions/safe/ffi/sys_uri';
+import { loadExtensions }  from './extensions';
```
Comment (joshuef) [inline app/background.process.development.js]: @hunterlester let's rename this to something w/ `background` in it.

`loadBackgroundExtensions`, maybe. Or `initExtensionsInBackground`... (other ideas welcome!)

Comment (joshuef) [reply app/background.process.development.js]: hmmm, actually maybe something clearer for the extension, `onInitBgProcess` makes most sense

File: app/extensions/safe/network/authenticator-comms.js [hunk 6]
```diff
@@ -14,6 +14,11 @@ let peruseAppObj;
 let store;
 let browserAuthReqUri;
 
+//////////////////////////
+//
+// QUESTION: Many of these functions are duplicated in the\
+// index of this directory. What is the purpose of these?
```
Comment (joshuef) [inline app/extensions/safe/network/authenticator-comms.js]: Aha, yeh. This is a hangover from me trying to find the 'correct' place for all of this. Was a bit awkward with some functionality in the main process, some in bg process etc.

So, the question probably should be... what's communicating here (the peruse app or webFetch? Are they going to be the same thing?).

Proooobably we should port the peruseApp functionality into the extension also (as that's SAFE specific, not browser generic.) So this file would be used by that app (be it registered or unregistered). 

Something like that... @hunterlester I'm open to suggestion! Now you've had a poke around and can see what's going on let's have a chat about this and what might be the best way to organise it.

File: app/setupPreloadAPIs.js [hunk 7]
```diff
@@ -119,6 +119,15 @@ const setupPreloadedSafeAuthAPIs = ( store ) =>
 
             if ( theCall.done && callPromises.resolve )
             {
+                if ( theCall.name === 'login' )
+                {
+                    logger.info('store subscribe calls: ', calls);
+                    logger.info('pendingCalls: ', pendingCalls);
+                    logger.info('call Promises: ', callPromises);
+		    // QUESTION: callPromises.resolve logs `null` \
```
Comment (joshuef) [inline app/setupPreloadAPIs.js:127]: Gooood Q. It should be....

Does this happen all the time? With a specific call? If we're not getting into the above condition then something's wrong for sure.

Maybe we should just be checking for `pendingCalls[theCall.id]`... although not sure if that'll make a difference.

**Comment by joshuef** [comment]
@hunter this is working well for me on OSX, so let's take a look at the couple of Qs re: file organisation and API hooks within the browser and then we can look at getting this merged 👍  nice one! 💥

File: app/setupPreloadAPIs.js (hunk 7 above)
Comment (hunterlester) [reply app/setupPreloadAPIs.js:127]: Yeah, the condition is met every time I login to authenticator.

I just don't understand how `theCall.done && callPromises.resolve` is met when `callPromises.resolve` is always null.

File: app/background.process.development.js (hunk 4 above)
Comment (hunterlester) [reply app/background.process.development.js]: Perhaps `isMock` should be in both `peruseApp` and `authenticator` states since each is associated with a respective native libarary?
Possibly useful if there were ever the case where `authenticator` was mock and `peruseApp` non-mock? Don't know.

File: app/extensions/safe/network/authenticator-comms.js (hunk 6 above)
Comment (hunterlester) [reply app/extensions/safe/network/authenticator-comms.js]: @joshuef Checkout latest commit. All of these functions relate to authenticator, so I moved them all to `authenticator-comms`, exporting them `network/index.js`.

File: app/background.process.development.js (hunk 4 above)
Comment (joshuef) [reply app/background.process.development.js]: Hmmm, I dont think that can happen, or? It's the auth that decides the network state not the app (which could auth against it in non-dev mode right?)

File: package.json [hunk 8]
```diff
@@ -7,7 +7,7 @@
   "description": "Electron web browser. React/redux powered with an aim of extensiblity.",
   "main": "main.js",
   "scripts": {
-    "test": "cross-env ELECTRON_RUN_AS_NODE=true NODE_ENV=test IS_UNPACKED=true BABEL_DISABLE_CACHE=1 electron ./test/runTests.js",
+    "test": "cross-env ELECTRON_RUN_AS_NODE=true NODE_ENV=test IS_UNPACKED=true BABEL_DISABLE_CACHE=1 node ./test/runTests.js",
```
Comment (hunterlester) [inline package.json]: Why the change? Tests run exactly the same here with `electron` except that `stdout` is properly piped when using `node` whereas tests run and pass with `electron` but am unable to see test output.

**Comment by joshuef** [comment]
closing as this is merged into #77
